src/main/java/org/jabref/logic/importer/fileformat/CffImporter.java

```diff
@@ -0,0 +1,155 @@
+import java.util.Map;
+
+import org.jabref.logic.importer.Importer;
+import org.jabref.logic.importer.ParserResult;
+import org.jabref.logic.util.StandardFileType;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.Field;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.entry.types.StandardEntryType;
+
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CffImporter extends Importer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CffImporter.class);
+
+    @Override
+    public String getName() {
+        return "CFF";
+    }
+
+    @Override
+    public StandardFileType getFileType() {
+        return StandardFileType.CFF;
+    }
+
+    @Override
+    public String getId() {
+        return "cff";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Importer for the CFF format. Is only used to cite software, one entry per file.";
+    }
+
+    private static class CffFormat {
+        private HashMap<String, String> vals = new HashMap<String, String>();
+
+        @JsonProperty("authors")
+        private List<CffAuthor> authors;
+
+        public CffFormat() {
+        }
+
+        @JsonAnySetter
+        private void setValues(String key, String value) {
+            vals.put(key, value);
+        }
+    }
+
+    private static class CffAuthor {
+        private HashMap<String, String> vals = new HashMap<String, String>();
+
+        public CffAuthor() {
+        }
+
+        @JsonAnySetter
+        private void setValues(String key, String value) {
+            vals.put(key, value);
+        }
+
+    }
+
+    @Override
+    public ParserResult importDatabase(BufferedReader reader) throws IOException {
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+        CffFormat citation = mapper.readValue(reader, CffFormat.class);
+        HashMap<Field, String> entryMap = new HashMap<Field, String>();
+
+        HashMap<String, StandardField> fieldMap = getFieldMappings();
+        for (Map.Entry<String, String> property : citation.vals.entrySet()) {
+            if (fieldMap.containsKey(property.getKey())) {
+                entryMap.put(fieldMap.get(property.getKey()), property.getValue());
+            }
+        }
+
+        List<String> authors = new ArrayList<String>();
+        for (CffAuthor auth: citation.authors) {
+            String aName = auth.vals.get("name");
+            String aGivenNames = auth.vals.get("given-names");
+            String aFamilyNames = auth.vals.get("family-names");
+            String aNameParticle = auth.vals.get("name-particle");
+            String aAlias = auth.vals.get("alias");
+
+            if (aName != null) {
+                authors.add(aName);
```

Comment:
JabRef has a class called Author which will handle the construction of authors, see org.jabref.model.entry
`    public Author(String first, String firstabbr, String von, String last, String jr) 
`
Afterwards, you put in an AuthorLIst  and return the and separated value
See for example the Medra-Fetcher  (org.jabref.logic.importer.fetcher)

Reply:
Apart from `String firstabbr`, CFF has the same fields for `person` type objects (e.g., authors). The main difference between these and the `entity` type objects (for this use case at least) is that entities only have a single field `name` instead, which I guess could be imported as is?

Reply:
- `first` -> [`given-names`](https://github.com/citation-file-format/citation-file-format/blob/1.2.0/schema-guide.md#definitionspersongiven-names)
- `von` -> [`name-particle`](https://github.com/citation-file-format/citation-file-format/blob/1.2.0/schema-guide.md#definitionspersonname-particle)
- `last` -> [`family-names`](https://github.com/citation-file-format/citation-file-format/blob/1.2.0/schema-guide.md#definitionspersonfamily-names)
- `jr` -> [`name-suffix`](https://github.com/citation-file-format/citation-file-format/blob/1.2.0/schema-guide.md#definitionspersonname-suffix)

Reply:
Yep, that seems to be correct, for reference:
https://github.com/JabRef/jabref/blob/cef4151c3805b080bf2a11b71d3182fdb5fd2446/src/main/java/org/jabref/model/entry/Author.java#L22-L31